### PR TITLE
remove misc deprecation warnings for 2.x

### PR DESCRIPTION
**Pull request by uraniumanchor**
# Contributing to the Donation Tracker

- [X] I've added tests or modified existing tests for the change.
- [X] I've humanly end-to-end tested the change by running an instance of the tracker.

### Issue from Pivotal Tracker

https://www.pivotaltracker.com/story/show/170941062

### Description of the Change

This one fixes the remaining 2.x runtime warnings that I could find.

The only behavior change is for how add/edit sets M2M relationships, and how the `set_lang` url is generated (though it's currently commented out, maybe I should have just deleted it).

### Verification Process

The only "user" facing behavior change is how runners are set on a run via the api, so I used a `fetch` in the console to make sure you could still write M2M this way.

```
formData = new FormData();
formData.append('type', 'run');
formData.append('id', 'nnn');
formData.append('runners', 'x,y');
fetch('/tracker/api/v1/edit/', {method: 'POST', credentials: 'same-origin', body: formData}).then(response => response.json()).then(json => console.log(json));
```

### CODE REVIEW

File: widgets.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def value_from_datadict(self, data, files, name):
         else:
             return None
 
-    def render(self, name, value, attrs=None):
+    def render(self, name, value, attrs=None, renderer=None):
```
Comment (faultyserver) [inline widgets.py]: I didn't see any calls to this change? Is this just something django wants?

File: tests/test_event.py [hunk 2]
```diff
@@ -40,10 +41,10 @@ def setUp(self):
         self.super_user = User.objects.create_superuser(
             'admin', 'admin@example.com', 'password'
         )
-        timezone = pytz.timezone('America/New_York')
+        timezone = pytz.timezone(settings.TIME_ZONE)
```
Comment (yunaa) [inline tests/test_event.py]: poggers

File: widgets.py (hunk 1 above)
Comment (yunaa) [reply widgets.py]: I'm similarly confused. I don't see it in the documentation.

Comment (yunaa) [reply widgets.py]: I guess this one: https://docs.djangoproject.com/en/dev/internals/deprecation/#deprecation-removed-in-2-1

> Support for Widget.render() methods without the renderer argument will be removed.

**Comment by uraniumanchor** [comment]
> LGTM. Interesting with the formatting changes in `urls.py`. Were those not `black`'d before? Or did our black config change?

Not sure, may have been another screwy rebase, because those changed last as part of the "auth views" branch.

File: widgets.py (hunk 1 above)
Comment (uraniumanchor) [reply widgets.py]: Yeah, it's deprecated and Django spits a warning about defining it. So off it went.

**Pull request merged**